Package "@angular-devkit/build-ng-packagr" has an incompatible peer dependency to "ng-packagr"

<!--🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅

Oh hi there! 😄

To expedite issue processing please search open and closed issues before submitting a new one.
Existing issues often contain information about workarounds, resolution, or progress updates.

🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅🔅-->


# 🐞 Bug report

### Command (mark with an `x`)
<!-- Can you pin-point the command or commands that are effected by this bug? -->
<!-- ✍️edit: -->

- [ ] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [ ] add
- [x] update
- [ ] lint
- [ ] extract-i18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc


### Is this a regression?

<!-- Did this behavior use to work in the previous version? -->
<!-- ✍️--> Yes, the previous version in which this bug was not present was: ....

Not sure.

### Description

<!-- ✍️--> A clear and concise description of the problem...

```
$ ng update @angular-devkit/build-angular @angular/cli @angular/core

                  Package "@angular-devkit/build-ng-packagr" has an incompatible peer dependency to "ng-packagr" (requires "^10.0.0", would install "11.0.2").
```


## 🌍 Your Environment
<pre><code>
<!-- run `ng version` and paste output below -->
<!-- ✍️-->
Angular CLI: 11.0.1
Node: 14.4.0
OS: win32 x64

Angular: 11.0.0
... animations, common, compiler, compiler-cli, core, elements
... forms, language-service, platform-browser
... platform-browser-dynamic, platform-server, router
Ivy Workspace: Yes

Package                         Version
---------------------------------------------------------
@angular-devkit/architect       0.1100.1
@angular-devkit/build-angular   0.1100.1
@angular-devkit/core            11.0.1
@angular-devkit/schematics      11.0.1
@angular/cli                    11.0.1
@schematics/angular             11.0.1
@schematics/update              0.1100.1
ng-packagr                      11.0.2
rxjs                            6.6.3
typescript                      4.0.5
webpack                         4.44.2
</code></pre>

When doing `ng update @angular/core @angular/cli`, I'm getting 

> Package "@angular-devkit/build-angular" has an incompatible peer dependency to "karma" (requires "~5.1.0", would install "5.2.3")
                  Package "@angular-devkit/build-ng-packagr" has an incompatible peer dependency to "ng-packagr" (requires "^10.0.0", would install "11.0.2").
× Migration failed: Incompatible peer dependencies found.

If I do `ng update @angular/core @angular/cli --force`, then I get

> An unhandled exception occurred: Cannot find module '@angular-devkit/core'
Require stack:
C:\Users\username\source\repos\Project\Project.Angular\node_modules\@angular\core\schematics\utils\project_tsconfig_paths.js
C:\Users\username\source\repos\Project\Project.Angular\node_modules\@angular\core\schematics\migrations\missing-injectable\index.js
C:\Users\username\AppData\Local\Temp\angular-cli-packages-qgcUmo\node_modules\@angular\cli\node_modules\@angular-devkit\schematics\tools\export-ref.js
C:\Users\username\AppData\Local\Temp\angular-cli-packages-qgcUmo\node_modules\@angular\cli\node_modules\@angular-devkit\schematics\tools\index.js
C:\Users\username\AppData\Local\Temp\angular-cli-packages-qgcUmo\node_modules\@angular\cli\utilities\json-schema.js
C:\Users\username\AppData\Local\Temp\angular-cli-packages-qgcUmo\node_modules\@angular\cli\models\command-runner.js
C:\Users\username\AppData\Local\Temp\angular-cli-packages-qgcUmo\node_modules\@angular\cli\lib\cli\index.js
C:\Users\username\AppData\Local\Temp\angular-cli-packages-qgcUmo\node_modules\@angular\cli\lib\init.js
C:\Users\username\AppData\Local\Temp\angular-cli-packages-qgcUmo\node_modules\@angular\cli\bin\ng
See "C:\Users\username\AppData\Local\Temp\ng-twMExu\angular-errors.log" for further details.

angular-errors.log
----------
[error] Error: Cannot find module '@angular-devkit/core'
Require stack:
C:\Users\username\source\repos\Project\Project.Angular\node_modules\@angular\core\schematics\utils\project_tsconfig_paths.js
C:\Users\username\source\repos\Project\Project.Angular\node_modules\@angular\core\schematics\migrations\missing-injectable\index.js
C:\Users\username\AppData\Local\Temp\angular-cli-packages-qgcUmo\node_modules\@angular\cli\node_modules\@angular-devkit\schematics\tools\export-ref.js
C:\Users\username\AppData\Local\Temp\angular-cli-packages-qgcUmo\node_modules\@angular\cli\node_modules\@angular-devkit\schematics\tools\index.js
C:\Users\username\AppData\Local\Temp\angular-cli-packages-qgcUmo\node_modules\@angular\cli\utilities\json-schema.js
C:\Users\username\AppData\Local\Temp\angular-cli-packages-qgcUmo\node_modules\@angular\cli\models\command-runner.js
C:\Users\username\AppData\Local\Temp\angular-cli-packages-qgcUmo\node_modules\@angular\cli\lib\cli\index.js
C:\Users\username\AppData\Local\Temp\angular-cli-packages-qgcUmo\node_modules\@angular\cli\lib\init.js
C:\Users\username\AppData\Local\Temp\angular-cli-packages-qgcUmo\node_modules\@angular\cli\bin\ng
    at Function.Module._resolveFilename (internal/modules/cjs/loader.js:623:15)
    at Function.Module._load (internal/modules/cjs/loader.js:527:27)
    at Module.require (internal/modules/cjs/loader.js:681:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at C:\Users\username\source\repos\Project\Project.Angular\node_modules\@angular\core\schematics\utils\project_tsconfig_paths.js:20:20
    at C:\Users\username\source\repos\Project\Project.Angular\node_modules\@angular\core\schematics\utils\project_tsconfig_paths.js:10:17
    at Object.<anonymous> (C:\Users\username\source\repos\Project\Project.Angular\node_modules\@angular\core\schematics\utils\project_tsconfig_paths.js:16:3)
    at Module._compile (internal/modules/cjs/loader.js:774:30)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:785:10)
    at Module.load (internal/modules/cjs/loader.js:641:32)
    at Function.Module._load (internal/modules/cjs/loader.js:556:12)
    at Module.require (internal/modules/cjs/loader.js:681:19)
    at require (internal/modules/cjs/helpers.js:16:16)
    at C:\Users\username\source\repos\Project\Project.Angular\node_modules\@angular\core\schematics\migrations\missing-injectable\index.js:22:38
    at C:\Users\username\source\repos\Project\Project.Angular\node_modules\@angular\core\schematics\migrations\missing-injectable\index.js:10:17
    at Object.<anonymous> (C:\Users\username\source\repos\Project\Project.Angular\node_modules\@angular\core\schematics\migrations\missing-injectable\index.js:16:3)



ng --version
-------------
Angular CLI: 10.2.0
Node: 12.3.1
OS: win32 x64

Angular: 10.2.3
... animations, common, compiler, compiler-cli, core, forms
... localize, platform-browser, platform-browser-dynamic, router
Ivy Workspace: Yes

Package                            Version
------------------------------------------------------------
@angular-devkit/architect          0.1002.0
@angular-devkit/build-angular      0.1002.0
@angular-devkit/build-ng-packagr   0.1002.0
@angular-devkit/core               10.2.0
@angular-devkit/schematics         10.2.0
@angular/cli                       10.2.0
@schematics/angular                10.2.0
@schematics/update                 0.1002.0
ng-packagr                         10.1.2
rxjs                               6.6.3
typescript                         3.9.7

--- a/packages/schematics/update/update/index.ts
+++ b/packages/schematics/update/update/index.ts
@@ -172,6 +172,12 @@ function _validateReversePeerDependencies(
         continue;
       }
 
+      if (installed === '@angular-devkit/build-ng-packagr') {
+        // Ignore peerDependencies mismatches for `@angular-devkit/build-ng-packagr`.
+        // This package is deprecated and is removed via a migration.
+        continue;
+      }
+
       // Override the peer version range if it's known as a compatible.
       const extendedRange = _updatePeerVersion(infoMap, peer, range);
 

